v3/testcases_test.go

```diff
@@ -0,0 +1,158 @@
+package geohex
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+)
+
+// Test cases downloaded from http://geohex.net/testcase/v3.2.html
+
+type code2LLTestCase struct {
+	code       string
+	expectedLL LL
+}
+
+func loadCode2LLTestCases() []code2LLTestCase {
+	var tcs []code2LLTestCase
+	// http://geohex.net/testcase/hex_v3.2_test_code2HEX.json
+	loadTestCasesFromJson("hex_v3.2_test_code2HEX.json", func(raw []json.RawMessage) error {
+		tc := code2LLTestCase{}
+		err := unmarshalRawFields([]tcFieldMapping{
+			{raw[0], &tc.code},
+			{raw[1], &tc.expectedLL.Lat},
+			{raw[2], &tc.expectedLL.Lon},
+		})
+		tcs = append(tcs, tc)
+		return err
+	})
+	return tcs
+}
+
+type ll2codeTestCase struct {
+	level        int
+	ll           LL
+	expectedCode string
+}
+
+func loadLL2CodeTestCases() []ll2codeTestCase {
+	var tcs []ll2codeTestCase
+	// http://geohex.net/testcase/hex_v3.2_test_coord2HEX.json
+	loadTestCasesFromJson("hex_v3.2_test_coord2HEX.json", func(raw []json.RawMessage) error {
+		tc := ll2codeTestCase{}
+		err := unmarshalRawFields([]tcFieldMapping{
+			{raw[0], &tc.level},
+			{raw[1], &tc.ll.Lat},
+			{raw[2], &tc.ll.Lon},
+			{raw[3], &tc.expectedCode},
+		})
+		tcs = append(tcs, tc)
+		return err
+	})
+	return tcs
+}
+
+type ll2PositionTestCase struct {
+	level     int
+	ll        LL
+	expectedX int
+	expectedY int
+}
+
+func loadLL2PositionTestCases() []ll2PositionTestCase {
```

Comment:
loading test cases is something you could do once in a https://onsi.github.io/ginkgo/#global-setup-and-teardown-beforesuite-and-aftersuite block. you should handle errors too